Devscord.DiscordFramework/Delegates.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Threading.Tasks;
+using Devscord.DiscordFramework.Framework;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Middlewares.Factories;
+using Discord.WebSocket;
+
+namespace Devscord.DiscordFramework
+{
+    public abstract class Delegates
+    {
+        public Func<Contexts, Task> UserJoined { get; set; }
+
+        protected Delegates()
+        {
+            Server.UserJoined += CallUserJoined;
```

Comment:
zmieniłbym to na odpalanie foreach listy Func<SocketGuildUser,Task> lub Action<SocketGuildUser>

wtedy prościej nad tym panować niż nad delegadami z += i -= 

powinno być jedno += które odpala foreach 
foreach może być w funkcji anonimowej 

+= user => userJoinedFunctions.ForEach(func => func.Invoke(user))

lub w wersji z klamrami i klasycznym foreach()

Reply:
a, jeszcze CallUserJoined powinno być uruchamiane przed tym 
i powinno to być coś typu GetUserContexts 
i powinno zwracać Contexts

Reply:
Nie za bardzo rozumiem, co ma zawierać lista userJoinedFunctions i jak ma się do niej dodawać kolejne funkcje?

Reply:
lista ma zawierać funkcje 
funkcje można trzymać w typach func i action 
(action to void a func coś zwraca)